features/nvstore/source/nvstore.h

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NVSTORE_H
+#define __NVSTORE_H
+
+// These addresses need to be configured according to board (in mbed_lib.json)
+#if defined(NVSTORE_AREA_1_ADDRESS) && defined(NVSTORE_AREA_1_SIZE) && \
+    defined(NVSTORE_AREA_2_ADDRESS) && defined(NVSTORE_AREA_2_SIZE) && \
+    defined(DEVICE_FLASH)
+#define NVSTORE_ENABLED 1
+#endif
+
+#if NVSTORE_ENABLED
+#include <stdint.h>
+#include <stdio.h>
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+
+enum {
```

Comment:
shall this have a name, and return types in the NVStore to be type of this enumeration ??

also limit the scope to be in the class?

Reply:
Wanted to keep it as simple as possible, having all functions returning int (as it's done in some of the block devices code). Is there any common practice here (int vs. enum)?